cpp/src/parquet/file_reader.cc

```diff
@@ -351,8 +364,11 @@ class SerializedFile : public ParquetFileReader::Contents {
     cached_source_ =
         std::make_shared<::arrow::io::internal::ReadRangeCache>(source_, ctx, options);
     std::vector<::arrow::io::ReadRange> ranges;
+    prebuffered_column_chunks_.clear();
```

Comment:
After double check, I think there could be concurrent issues. If `PreBuffer` and `GetRowGroup` can be called concurrently, `cached_source_` and `prebuffered_column_chunks_` might be in an inconsistent state. For example, `GetRowGroup` reads an old state in `prebuffered_column_chunks_`, then `PreBuffer` creates brand new `cached_source_` and `prebuffered_column_chunks_`. After that, `GetRowGroup` may pass the new `cached_source_` to the RowGroupReader. I believe the followup read will get an error complaining `Status::Invalid("ReadRangeCache did not find matching cache entry")`.

cc @westonpace

Reply:
That sounds reasonable. Thanks!